### PR TITLE
Changed to get this working at parts.sudoroom.org

### PR DESCRIPTION
**Pull request by Juul**
I added a simple startup script and made it possible to use with the `certbot` auto-renew feature by allowing arbitrary files to be served up at the root URL from the `static/webroot` directory.

### CODE REVIEW

**Comment by Juul** [comment]
I also had to do [this](https://github.com/Juul/stuff-org/commit/5f8df6af9542f91c48bf60f61ef7871d0cb13293) to get it working on our server.

File: startup_scripts/stuff-org [hunk 1]
```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
```
Comment (hzeller) [inline startup_scripts/stuff-org]: If we have a systemd stuff-org.service, we probably shouldn't do this via the detour of a sysV init script that then is invoked by systemd, but rather have systemd start the service directly.

File: stuff/image-handler.go [hunk 2]
```diff
@@ -38,10 +38,14 @@ func AddImageHandler(store StuffStore, template *TemplateRenderer, imgPath strin
...
-	// static handler.
-	http.HandleFunc("/robots.txt", func(w http.ResponseWriter, r *http.Request) {
-		sendResource(staticPath+"/robots.txt", "", w)
+	// With serving robots.txt and other static files,
+	// image-handler should probably be named static handler.
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		if(strings.EqualFold("/", r.URL.Path)) {
+			http.Redirect(w, r, "/search", 302);
+		} else {
+			sendResource(staticPath+"/webroot/"+r.URL.Path[1:], "", w)
```
Comment (hzeller) [inline stuff/image-handler.go]: From what I understand, you would like to have anything that is relative to `/` and not specific to the application (such as /static or /search), just come out of /static/webroot. I guess you can make this work if you also move the file currently in `/stuff/static/robots.txt` into the new directory `/stuff/static/webroot/robots.txt`.

File: stuff/search-handler.go [hunk 3]
```diff
@@ -158,23 +157,23 @@ func (h *SearchHandler) apiSearchPageItem(out http.ResponseWriter, r *http.Reque
 		Items:      make([]JsonHtmlSearchResultRecord, outlen),
 	}
 
-	pusher, _ := out.(http.Pusher) // HTTP/2 pushing if available.
+//	pusher, _ := out.(http.Pusher) // HTTP/2 pushing if available.
```
Comment (hzeller) [inline stuff/search-handler.go]: Please don't include this in the pull request for upstream.
You probably need to comment that out in a branch locally if you run an old golang.

**Comment by hzeller** [comment]
Thanks Marc!
I think the `systemd` thing can be simplified by putitng everything in the service definition without the sysV detour (someone who knows systemd can probably help).

For the webroot thing, you also need to git mv the robots.txt to the new place.

**Comment by hzeller** [comment]
(it is worthwhile to install a fresh go, because serving the images will be much faster)

**Comment by SuperQ** [comment]
This is probably better handled by putting a reverse proxy in front of stuff-org. Caddy, nginx, etc.

**Comment by hzeller** [comment]
I am curious: how to reverse proxies with http2 ? Ideally, we'd like to deliver the images quicker using an HTTP2 feature, but it is SSL only.

So with a reverse proxy, we provide the external SSL endpoint, but then also SSL talk to the internal golang service again. So we need to add and remove SSL 

How is this done in an easy way without too much administrative overhead ? I suppose the proxy and the internal server just use a self-signed certificate ? Or how is this done these days ?

**Comment by SuperQ** [comment]
Usually we just talk http2 over localhost with no crypto. Or, like you say, use a local CA.

**Comment by hzeller** [comment]
wasn't part of http2 that it _always_ is with crypto?

**Comment by SuperQ** [comment]
https://en.wikipedia.org/wiki/HTTP/2#Encryption

**Comment by hzeller** [comment]
lol, I guess I just had only the de-facto standard internalized

**Comment by hzeller** [comment]
https://godoc.org/golang.org/x/net/http2/h2c : seems to be the package to have golang actually work to do this via nonTLS, if someone wants to add a flag to stuff-org to be used in this proxy context.

(but, I'd actually just simply run the go server directly on that domain, it makes it simple and straightforward without much fuss. So I still like the approach @Juul adds.)

**Comment by sunjam** [comment]
It'd be interesting to get this going so one search could be placed against both the Noisebridge and Sudoroom databases.

**Comment by hzeller** [comment]
So it should be possible to send out the same query to both backends (which result in some JSON output), and then have a combined output somewhere, There might be web-detail things such as same origin that someone who know the details about these things will have to resolve.

**Pull request closed without merging**